### PR TITLE
deep copy complex command

### PR DESCRIPTION
**Pull request by Thomson19**

### CODE REVIEW

**Comment by Thomson19** [comment]
Tomasz Siekacz 215873
Bartłomiej Górkiewicz 215727

File: src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-public interface ICompoundCommand extends DriverCommand {
+public interface ICompoundCommand extends DriverCommand, Iterable<DriverCommand> {
 
-	public Iterator<DriverCommand> iterator();
+	Iterator<DriverCommand> iterator();
 
+	@Override
+	default DriverCommand clone() throws CloneNotSupportedException{
+		List<DriverCommand> commands = new ArrayList<>();
+		this.forEach(commands::add);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java:17]: I am not sure that this is enough. You only copy references to existing commands (without making a copy). This not how cloning should look.

File: src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java [hunk 2]
```diff
@@ -28,31 +29,12 @@ public synchronized void setCurrentCommand(DriverCommand commandList) {
...
-				return driverCommands.iterator();
-			}
-
-			@Override
-			public String toString() {
-				return name;
-			}
-		});
-
+		setCurrentCommand(new CompoundCommand(commandList));
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java:37]: Watch out: there might be conflicts with other merges (it depends on who will merge first ;) )

File: src/test/java/edu/kis/powp/jobs2d/events/SelectDeepCopySecretCommand.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            commands.add(new OperateToCommand(0, -50));
+            commands.add(new SetPositionCommand(0, -40));
+            commands.add(new OperateToCommand(0, 50));
+            commands.add(new SetPositionCommand(70, -50));
+            commands.add(new OperateToCommand(20, -50));
+            commands.add(new OperateToCommand(20, 0));
+            commands.add(new OperateToCommand(70, 0));
+            commands.add(new OperateToCommand(70, 50));
+            commands.add(new OperateToCommand(20, 50));
+
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectDeepCopySecretCommand.java:36]: To test DEEP copy, you should have compound command embedded in another compound command.

**Review by kowallus — CHANGES_REQUESTED** [review]
Check the issues, close PR and reopen after fixes.

**Pull request closed without merging**